src/Estimators/src/BaseEstimatorFromFootIMU.cpp

```diff
@@ -0,0 +1,293 @@
+        thetaX = 0;
+    }
+
+    return output;
+}
+
+using namespace BipedalLocomotion::Estimators;
+
+bool BaseEstimatorFromFootIMU::setModel(const iDynTree::Model& model)
+{
+    constexpr auto logPrefix = "[BaseEstimatorFromFootIMU::setModel]";
+
+    m_isModelSet = false;
+
+    m_model = model;
+
+    if (!m_kinDyn.loadRobotModel(model))
+    {
+        log()->error("{} Unable to load the model.", logPrefix);
+        return false;
+    }
+
+    m_isModelSet = true;
+
+    return true;
+}
+
+bool BaseEstimatorFromFootIMU::initialize(
+    std::weak_ptr<const ParametersHandler::IParametersHandler> handler)
+{
+    constexpr auto logPrefix = "[BaseEstimatorFromFootIMU::initialize]";
+
+    m_isInitialized = false;
+
+    if (!m_isModelSet)
+    {
+        log()->error("{} No iDynTree::Model has been set.", logPrefix);
+        return false;
+    }
+
+    if (!m_kinDyn.isValid())
+    {
+        log()->error("{} iDynTree::KinDynComputations m_kinDyn is not valid.", logPrefix);
+        return false;
+    }
+
+    auto ptr = handler.lock();
+    if (ptr == nullptr)
+    {
+        log()->error("{} Invalid parameter handler.", logPrefix);
+        return false;
+    }
+
+    auto populateparameter
+        = [logPrefix](std::shared_ptr<const ParametersHandler::IParametersHandler> ptr,
+                      const std::string& paramName,
+                      auto& parameter) -> bool {
+        if (!ptr->getParameter(paramName, parameter))
+        {
+            log()->error("{} Unable to find the parameter '{}'", logPrefix, paramName);
+            return false;
+        }
+        return true;
+    };
+
+    bool ok = populateparameter(ptr->getGroup("MODEL_INFO").lock(),
+                                "base_frame",
+                                m_baseFrameName); /**< root link of the robot (whose
+                                                   pose must be estimated) */
+    ok = populateparameter(ptr->getGroup("MODEL_INFO").lock(),
+                           "foot_frame",
+                           m_footFrameName); /**< foot frame of the robot (whose
+                                              orientation is measured) */
+    ok = ok && populateparameter(ptr, "foot_width_in_m", m_footWidth);
+    ok = ok && populateparameter(ptr, "foot_length_in_m", m_footLength);
+
+    // Set the 4 foot vertices in World reference frame [dimensions in meters]
+    m_cornersInLocalFrame.emplace_back(+m_footWidth / 2, +m_footLength / 2, 0);
+    m_cornersInLocalFrame.emplace_back(-m_footWidth / 2, +m_footLength / 2, 0);
+    m_cornersInLocalFrame.emplace_back(-m_footWidth / 2, -m_footLength / 2, 0);
+    m_cornersInLocalFrame.emplace_back(+m_footWidth / 2, -m_footLength / 2, 0);
+
+    m_gravity << 0, 0, -9.81; // TO DO: verify the reference frame
+    m_frameIndex = m_kinDyn.getFrameIndex(m_footFrameName); // TO DO: set the proper name
+                                                            // depending on the stance foot.
```

Comment:
In theory you can use two objects of this class s no need to set it depending on the stance foot